lib/Frontend/CompilerInvocation.cpp

```diff
@@ -306,11 +304,14 @@ static bool ParseCodeGenArgs(CodeGenOptions &Opts, ArgList &Args, InputKind IK,
   bool Success = true;
 
   unsigned OptLevel = getOptimizationLevel(Args, IK, Diags);
-  if (OptLevel > 3) {
-    Diags.Report(diag::err_drv_invalid_value)
-      << Args.getLastArg(OPT_O)->getAsString(Args) << OptLevel;
-    OptLevel = 3;
-    Success = false;
+  unsigned MaxOptLevel = 3;
+  if (IK == IK_OpenCL){
```

Comment:
There is no need in upstreaming patches from spir1.2 branch, since all of them should be already in spirv branch. This particular patch is a workaround for the issue #3. 
This issue is already resolved in the trunk with 'noduplicate' attribute, so no need to apply this workaround to the trunk.
Does it make sense to you?